include/boost/math/differentiation/autodiff.hpp

```diff
@@ -101,6 +106,8 @@ class fvar
     template<typename RealType2>
     fvar(const RealType2& ca); // Supports any RealType2 for which static_cast<root_type>(ca) compiles.
 
+    explicit fvar(const char* ca); // Converts a const char* string to RealType by boost::lexical_cast
```

Comment:
Is this constructor necessary? Would prefer to do:

```c++
autodiff_fvar<double,5> x(boost::lexical_cast<double>("3.14159"));
```
Not that it's likely, but some AD implementations I've seen allow one to *name* the variables. So by adding this, we would close ourselves off from this possibility:

```c++
autodiff_fvar<double,5> x("x");
```

Reply:
During static initialization, `unchecked_factorial` (actually `unchecked_factorial_imp`) calls `boost::math::tools::convert_from_string<T>(const char*)`, which returns either `const char*` or type `T` (depending on whether `T` is constructible from `const char*`). Without the constructor, we'll need a way to have `boost::lexical_cast` (called by `boost::math::tools::convert_from_string`) operate on the root type rather than on autodiff_fvar directly. The extra code in the constructor is due to `float` causing `boost::lexical_cast` to throw an error when the factorials are too large too fit within the `float` type.

I agree, though: it's not an ideal situation.